wfe/wfe_test.go

```diff
@@ -942,6 +942,27 @@ func TestChallenge(t *testing.T) {
 	test.AssertEquals(t, responseWriter.Code, http.StatusNotFound)
 	assertJSONEquals(t, responseWriter.Body.String(),
 		`{"type":"urn:acme:error:malformed","detail":"Expired authorization","status":404}`)
+
+	// Challenge Not found
+	challengeURL = "not_found/23"
+	responseWriter = httptest.NewRecorder()
+	wfe.Challenge(ctx, newRequestEvent(), responseWriter,
+		makePostRequestWithPath(challengeURL,
+			signRequest(t, `{"resource":"challenge"}`, wfe.nonceService)))
+	test.AssertEquals(t, responseWriter.Code, http.StatusNotFound)
+	assertJSONEquals(t, responseWriter.Body.String(),
+		`{"type":"urn:acme:error:malformed","detail":"No such challenge","status":404}`)
+
+	// When an error different than sql.ErrNoRows is returned
```

Comment:
Looking at this again I think I understand the source of my confusion now and have a suggestion. Right now this comment says that it's for an "error different than sql.ErrNoRows is returned" which is true, but the `mocks/mocks.go` implementation of `GetAuthorization`'s default return is conceptually similar to a ErrNoRows in that it returns `core.Authorization{}, fmt.Errorf("authz not found")`. 

I think it might be less confusing if you removed your diff to `mocks/mocks.go` that returns `sql.ErrNoRows` for id "not_found" and instead make that the default return value. Then you can add a similar diff but for id "error_result" that returns something like `fmt.Errorf("Unspecified database error")`. Then this test can assume any ID that doesn't exist will return `sql.ErrNoRows` and the specific id `error_result` will test the non sql.ErrNoRows code path.

Does that make sense?